src/processor.jl

```diff
@@ -310,3 +316,6 @@ rmprocs!(ctx::Context, xs::AbstractVector{<:Integer}) = rmprocs!(ctx, map(OSProc
 rmprocs!(ctx::Context, xs::AbstractVector{<:Processor}) = lock(ctx) do
     filter!(p -> p ∉ xs, ctx.procs)
 end
+
+"Gets the current processor executing the current thunk."
+thunk_processor() = task_local_storage(:processor)
```

Comment:
Maybe add a type check to make this type stable? Or is it possible to be many different kind of Processors? Even checking that it is a subtype of an Abstact processor might be helpful.

Reply:
Yeah we can only really assert `::Processor`. I just pushed the typeassert.